### PR TITLE
Document .gitignore and .odoignore usage

### LINKED ISSUE #6461: Document odo dev behavior with gitignore and odoignore
/area documentation

/kind bug

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind feature
/area documentation

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6461 

Preview:
- https://deploy-preview-6745--odo-docusaurus-preview.netlify.app/docs/overview/configure#ignoring-files-to-synchronize
- https://deploy-preview-6745--odo-docusaurus-preview.netlify.app/docs/development/architecture/how-odo-works#project-sources

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: docs/website/docs/development/architecture/how-odo-works.md [hunk 1]
```diff
@@ -732,12 +732,13 @@ spec:
...
 
-However, this is subject to two things:
+However, this is subject to three things:
 - the value of the `mountSources` flag (default value is `true`) in the Devfile container component. Project sources are not mounted in the container if this is set to `false`.
   Note that odo requires at least one component in the Devfile to set `mountSources: true` in order to synchronize files.
 - the type of volume created depends on the [configuration of `odo`](../../overview/configure#preference-key-table), and more specifically on the value of the `Ephemeral` setting:
   - if `Ephemeral` is `false`, which is the default setting, `odo` creates a [PersistentVolumeClaim](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims) (PVC) (with the default storage class)
   - if `Ephemeral` is `true`, `odo` creates an [`emptyDir`](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) volume, tied to the lifetime of the Pod.
+- the complete content of the current directory and its sub-directories is synchronized to the container, except the files listed in the `.odoignore` file, or, if this file is not present, in the `.gitignore` file.
```
Comment (rm3l) [inline docs/website/docs/development/architecture/how-odo-works.md:741]: Maybe we could also add a link to the other page, as it provides additional details?

File: docs/website/docs/overview/configure.md [hunk 2]
```diff
@@ -172,6 +172,19 @@ You can use the `--force` (or `-f`) flag to force the deletion of the registry w
 To update a registry, you can delete it and add it again with the updated value.
 :::
 
+## Ignoring files to synchronize
```
Comment (rm3l) [inline docs/website/docs/overview/configure.md:175]: Wondering why this is under "Configuration", as, to me, it does not relate to the `odo` configuration, but depends on each project..
Perhaps a dedicated page for this (under "User Guides" or "User Guides > Advanced Usage") would make more sense, no?

Comment (feloy) [reply docs/website/docs/overview/configure.md:175]: Or in the page "Only Push Specific Files" in "User Guides > Advanced Usage". WDYT @rm3l ?

Comment (rm3l) [reply docs/website/docs/overview/configure.md:175]: Ah yes, good point. That would be better, indeed. Thanks.

**Review by rm3l — APPROVED** [review]

**Comment by feloy** [comment]
/override Kubernetes-Integration-Tests/Kubernetes-Integration-Tests
/override OpenShift-Integration-tests/OpenShift-Integration-tests

Doc only

**Pull request merged**